Doc/reference/ldap-modlist.rst

```diff
@@ -19,6 +19,14 @@ The :mod:`ldap.modlist` module defines the following functions:
    names which shall be ignored completely. Attributes of these types will not appear
    in the result at all.
 
+   The value parts of the *entry* dictionary must be any of a list of
+   :py:class:`bytes` objects or a :py:class:`bytes` object (in Python 3).
```

Comment:
I start to think that accepting non-list `bytes` objects is possibly a bug.
The code below seems to assume values of entry dicts (the `value` variable in the code) are `list` (or `tuple`), not `bytes`.
Resultant `attrvaluelist` is always a list.
`attrvaluelist` is however never used thereafter and the original `value` variable is used instead.

https://github.com/python-ldap/python-ldap/blob/master/Lib/ldap/modlist.py#L20-L23

Reply:
To me, the intended behavior for single `bytes` (and for `None items) it's also a bit of a mystery. Let's be careful here, to not break users' code.